VespaGAS.xcodeproj/project.pbxproj

```diff
@@ -8,6 +8,7 @@
 
```

Comment:
>  you need to explicitly say that a type conforms to Equatable, i.e., you also need to add : Equatable to the type declaration.

Both Gasoline and Oil conforms to Equatable because Volume does it in first instance. 
I don't see the value in specifying again for Gasoline and Oil that they are equatable. I could provide a custom implementation for both of them, but the result will be the same. 

We want to be sure that: 

1. 2 Volume objects are equal if they have the same amount and unit
1. 2 Gasoline objects are equal if they have the same amount and unit
1. 2 Oil objects are equals if they have the same amount and unit
1. Anything else is not equal: I can't even compare Oil with Gasoline with this implementation